src/ledger/mod.rs

```diff
@@ -322,6 +324,76 @@ impl Ledger {
         Ok(())
     }
 
+    #[instrument(name = "ledger.batch_dropped", skip(self, tx))]
+    pub async fn batch_dropped(
+        &self,
+        tx: Transaction<'_, Postgres>,
+        tx_id: LedgerTransactionId,
+        created_batch_tx_id: LedgerTransactionId,
+    ) -> Result<(), LedgerError> {
+        let txs = self
+            .inner
+            .transactions()
+            .list_by_ids(std::iter::once(created_batch_tx_id))
+            .await?;
+        let txn = txs.first().ok_or(LedgerError::TransactionNotFound)?;
+
+        let BatchCreatedMeta {
+            batch_info,
+            tx_summary,
+        } = txn.metadata()?.ok_or(LedgerError::MissingTxMetadata)?;
+
+        let entries = self
+            .inner
+            .entries()
+            .list_by_transaction_ids(std::iter::once(created_batch_tx_id))
+            .await?;
+
+        let mut ledger_account_ids = WalletLedgerAccountIds::default();
+        let mut encumbered_fees: Satoshis = Satoshis::from_btc(Decimal::ZERO);
+        for entry in entries.into_values().flatten() {
+            match entry.entry_type.as_str() {
+                "BATCH_CREATED_LOG_OUT_ENC_DR" => {
+                    ledger_account_ids.effective_outgoing_id = entry.account_id;
+                }
+                "BATCH_CREATED_LOG_SET_DR" => {
+                    ledger_account_ids.effective_at_rest_id = entry.account_id;
+                }
+                "BATCH_CREATED_FR_ENC_CR" => {
+                    ledger_account_ids.fee_id = entry.account_id;
+                    encumbered_fees = Satoshis::from_btc(entry.units);
+                }
+                "BATCH_CREATED_UTX_OUT_PEN_CR" => {
+                    ledger_account_ids.onchain_outgoing_id = entry.account_id;
+                }
+                "BATCH_CREATED_CHG_ENC_CR" => {
+                    ledger_account_ids.onchain_incoming_id = entry.account_id;
+                }
+                "BATCH_CREATED_UTX_SET_DR" => {
+                    ledger_account_ids.onchain_at_rest_id = entry.account_id;
+                }
+                _ => {}
```

Comment:
Are we sure this is exhaustive? Perhaps `_ => unimplemented!()` would be good as a sanity check.